extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java

```diff
@@ -170,9 +170,11 @@ public void run() {
         SecurityRealm realm = LegacyPropertiesSecurityRealm.builder()
                 .setDefaultRealm(config.realmName)
                 .setProviders(new Supplier<Provider[]>() {
+                    Provider[] providers = new Provider[] { new WildFlyElytronProvider() };
+
                     @Override
                     public Provider[] get() {
-                        return new Provider[] { new WildFlyElytronProvider() };
+                        return providers;
```

Comment:
Maybe you can make the whole supplier static and reuse it at both locations?

Reply:
done